xbmc/video/VideoInfoScanner.cpp

```diff
@@ -1561,12 +1567,14 @@ namespace VIDEO
       std::string aspect = url.m_aspect;
       if (aspect.empty())
         // Backward compatibility with Kodi 11 Eden NFO files
-        aspect = ContentToMediaType(content, pItem->m_bIsFolder) == MediaTypeEpisode ? "thumb" : "poster";
-      if (find(artTypes.begin(), artTypes.end(), aspect) == artTypes.end() || art.find(aspect) != art.end())
-        continue;
-      std::string image = GetImage(url, pItem->GetPath());
-      if (!image.empty())
-        art.insert(std::make_pair(aspect, image));
+        aspect = mediaType == MediaTypeEpisode ? "thumb" : "poster";
+      if ((addAll || find(artTypes.begin(), artTypes.end(), aspect) == artTypes.end()) &&
```

Comment:
The art type has to be in art types
` find(artTypes.begin(), artTypes.end(), aspect) != artTypes.end()`

Not really reviewing, but happened to notice this typo